VulkanLaunchpad.cpp

```diff
@@ -669,7 +670,7 @@ vk::MemoryAllocateInfo vklCreateMemoryAllocateInfo(vk::DeviceSize bufferSize, vk
         }
 
         // Does this kind of memory support our usage requirements?
-        if ((memoryProperties.memoryTypes[i].propertyFlags & (vk::MemoryPropertyFlagBits::eHostVisible | vk::MemoryPropertyFlagBits::eHostCoherent))
+        if ((memoryProperties.memoryTypes[i].propertyFlags & (flags))
```

Comment:
Ah, damn... I'm afraid there's a potential bug lurking (which has been lurking since forever 😱).
The problem is as follows:
- We only check if the flags are supported
- But there might be different heaps (of different size)
and that latter point might be a problem. See here:

![image](https://user-images.githubusercontent.com/536330/214881024-918daba7-cd80-4e14-8fc8-171d49be6235.png)

Both memory types have the `DEVICE_LOCAL_BIT` as property flag, but heap `[2]` is only 224 MB in size, while heap `[0]` is 8 GB large.

We should change the lambda which selects the the [`memoryTypeIndex`](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkMemoryAllocateInfo.html) so that it selects the memory type which fulfills both:
1. supports all the required flags
2. belongs to the largest heap (among those which support all required flags).

The different heap properties are contained in [`VkPhysicalDeviceMemoryProperties`](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkPhysicalDeviceMemoryProperties.html) (or `auto memoryProperties`  in the existing code).

Which heap a memory type belongs to is stored in [`VkMemoryType::heapIndex`](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkMemoryType.html) (or `memoryProperties.memoryTypes[i].heapIndex`).

While you're at it, could you change memory type selection according to 👆 this description?